Bottom: b7be02c28b0a823e4febc6d0fe6280aca0572320
Top:    c409939610256c83cdcf4b4ccec1db8ec5a38f1f
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-25 15:52:39 +0100

Refresh of draw-c-remove-updateall-api-in

---

diff --git a/src/action.c b/src/action.c
index 78df9fe..ed4ca24 100644
--- a/src/action.c
+++ b/src/action.c
@@ -2829,12 +2829,12 @@ ActionDisplay (int argc, char **argv, int childX, int childY)
 
 	case F_ToggleName:
 	  TOGGLE_FLAG (SHOWNUMBERFLAG, PCB);
-	  UpdateAll ();
+	  Redraw ();
 	  break;
 
 	case F_ToggleMask:
 	  TOGGLE_FLAG (SHOWMASKFLAG, PCB);
-	  UpdateAll ();
+	  Redraw ();
 	  break;
 
 	case F_ToggleClearLine:
@@ -2861,7 +2861,7 @@ ActionDisplay (int argc, char **argv, int childX, int childY)
 	  /* toggle displaying of the grid */
 	case F_Grid:
 	  Settings.DrawGrid = !Settings.DrawGrid;
-	  UpdateAll ();
+	  Redraw ();
 	  break;
 
 	  /* display the pinout of an element */
@@ -2970,7 +2970,7 @@ ActionDisplay (int argc, char **argv, int childX, int childY)
 	      PCB->GridOffsetX = atoi (argv[1]);
 	      PCB->GridOffsetY = atoi (argv[2]);
 	      if (Settings.DrawGrid)
-		UpdateAll ();
+		Redraw ();
 	    }
 	  break;
 
diff --git a/src/draw.c b/src/draw.c
index 10da1dd..7619d25 100644
--- a/src/draw.c
+++ b/src/draw.c
@@ -165,15 +165,6 @@ AddPart (void *b)
 }
 
 /*
- * force the whole output to be updated
- */
-void
-UpdateAll (void)
-{
-  gui->invalidate_all ();
-}
-
-/*
  * initiate the actual drawing to the pixmap/screen
  * make the update block slightly larger to handle round-off
  * caused by the TO_SCREEN operation
@@ -201,7 +192,7 @@ Draw (void)
 void
 ClearAndRedrawOutput (void)
 {
-  UpdateAll ();
+  Redraw ();
 }
 
 
diff --git a/src/draw.h b/src/draw.h
index c60b6d9..d62ed7d 100644
--- a/src/draw.h
+++ b/src/draw.h
@@ -72,7 +72,6 @@ void EraseElementPinsAndPads (ElementTypePtr);
 void EraseElementName (ElementTypePtr);
 void EraseObject (int, void *, void *);
 void LoadBackgroundImage (char *);
-void UpdateAll (void);
 
 /*GdkDrawable *draw_get_current_drawable(void);*/
 
diff --git a/src/set.c b/src/set.c
index 17f7583..90db719 100644
--- a/src/set.c
+++ b/src/set.c
@@ -82,7 +82,7 @@ SetGrid (double Grid, bool align)
 	}
       PCB->Grid = Grid;
       if (Settings.DrawGrid)
-	UpdateAll ();
+	Redraw ();
     }
 }
